main.go

```diff
@@ -26,44 +28,61 @@ import (
 )
 
 const (
-	pollInterval                = 15 * time.Second
-	pollTimeout                 = 3600 * time.Second
-	diskPath             string = "./tmp/disk.img.gz"
-	diskPathDecompressed string = "./tmp/disk.img"
-	diskPathConverted    string = "./tmp/disk.qcow2"
+	pollInterval               = 15 * time.Second
+	pollTimeout                = 3600 * time.Second
+	diskPathConverted   string = "./tmp/disk.qcow2"
+	certificatePath     string = "./tmp/tls.crt"
+	kvExportToken       string = "x-kubevirt-export-token"
+	kvExportTokenLength int    = 20
 )
 
-func applyVirtualMachineExport(client kubecli.KubevirtClient, vmNamespace, vmName string) error {
-	log.Println("Applying VirtualMachineExport object...")
+func createSecret(client kubecli.KubevirtClient, vmNamespace, vmName string) error {
+	token, err := generateSecureRandomString(kvExportTokenLength)
+	if err != nil {
+		return err
+	}
+
+	v1Secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vmName,
+			Namespace: vmNamespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			kvExportToken: []byte(token),
+		},
+	}
+	_, err = client.CoreV1().Secrets(vmNamespace).Create(context.Background(), v1Secret, metav1.CreateOptions{})
```

Comment:
I can't see anywhere command to delete the secret at the end of the task

Reply:
I wanted to delete secret and vmexport, but then I realized that the execution will be stopped if there is an error. Do you have an idea how to handle it?

Reply:
I thought to set an owner of the secret and vmexport, so if the kubevirt-disk-uploader pod is deleted, it deletes deployed resources by this owner pod.

Reply:
If the task would be in tekton tasks, you could reuse this package :D: https://github.com/kubevirt/kubevirt-tekton-tasks/blob/main/modules/shared/pkg/ownerreference/ownerreference.go
Maybe you can copy it (or import is as a package via go.mod) and when the task will be migrated to tekton tasks, you can remove it without any additional change

Reply:
@ksimon1 Thanks! I'll take a look.

I'll work on resolving this issue of ownership (aka deleting secret/vmexport) in this PR, because it affects the functionality of kubevirt-disk-uploader when you want to run it twice.